conans/client/tools/settings.py

```diff
@@ -28,8 +31,14 @@ def add_millennium(cppstd):
         rhs = add_millennium(extract_cpp_version(rhs))
         return lhs < rhs
 
+    def is_linux(conanfile):
+        os = conanfile.settings.get_safe("os_build")
```

Comment:
Still, the C++ standard is the host one, not the build one. This is why the OSInfo() cannot be used!